services/server/server.go

```diff
@@ -493,6 +514,67 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	return res, nil
 }
 
+func (s *Server) validateSSHKey(sshKey string) error {
+	tempFile := path.Join(os.TempDir(), "temp_keys")
+	err := ioutil.WriteFile(tempFile, []byte(sshKey), os.FileMode(0600))
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command("ssh-keygen", "-l", "-f", tempFile)
+	pdeathsig.Set(cmd, unix.SIGKILL)
+
+	err = cmd.Start()
+	if err != nil {
+		return errors.Wrap(err, "couldn't start ssh-keygen process")
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		if e, ok := err.(*exec.ExitError); ok && e.ExitCode() != 0 {
+			return status.Errorf(codes.InvalidArgument, "Invalid ssh key")
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (s *Server) writeSSHKey(sshKey string) error {
+	username := "admin"
+	usr, err := user.Lookup(username)
+	if err != nil {
+		return err
+	}
+	uid, err := strconv.Atoi(usr.Uid)
+	if err != nil {
+		return err
+	}
+	gid, err := strconv.Atoi(usr.Gid)
+	if err != nil {
+		return err
+	}
+	sshDirPath := path.Join(usr.HomeDir, ".ssh")
+	err = os.Mkdir(sshDirPath, os.FileMode(0700))
+	if err != nil && !os.IsExist(err) {
+		return err
+	}
+	err = os.Chown(sshDirPath, uid, gid)
+	if err != nil {
+		return err
+	}
+	keysPath := path.Join(sshDirPath, "authorized_keys")
+	err = ioutil.WriteFile(keysPath, []byte(sshKey), os.FileMode(0600))
```

Comment:
We should protect from concurrent calls with mutex.

Reply:
We've already protected it with mutex here https://github.com/percona/pmm-managed/pull/294/files#diff-6f891b9f28cfb10751fe182afa544a9cR428. Do we need extra one?

Reply:
`envRW` protects `envXXX` fields. That line acquires shared read lock. We need an exclusive lock for `authorized_keys` file.